CHANGELOG.md

```diff
@@ -1,7 +1,22 @@
 ## vNext (TBD)
 
 ### Enhancements
-* None
+* Added a stronger write and commit async API. This deprecates `Realm.WriteAsync(Action<Realm> action)`, `Real.WriteAsync<T>(Func<Realm, IQueryable<T>> function)`, `Realm.WriteAsync<T>(Func<Realm, IList<T>> function)` and `Realm.WriteAsync<T>(Func<Realm, T> function)`. The new API is made of `Realm.WriteAsync<T>(Func<T> function)`, `Realm.WriteAsync(Action action)`, `Realm.BeginWriteAsync()` and `Transaction.CommitAsync()`. While the `Transaction.Rollback()` stays unchanged as it does not need an async version. Importantly, you can mix and match async and sync calls. And the new WriteAsync API executes on the same thread where it's called, so you can capture Realms and RealmObjects/EmbeddedObjects in the delegate like so:
```

Comment:
I think we can reword this a little to be more explicit - e.g. "stronger" is a bit vague and the key difference of the API is left to the end sentence, while we should probably explain the distinction between the old and the new earlier in the text. I'm thinking something like:

```
* Added a more efficient replacement for `Realm.WriteAsync`. The previous API would start a background thread, open the Realm there and run a synchronous write transaction on the background thread. The new API will asynchronously acquire the write lock (begin transaction) and asynchronously commit the transaction, but the actual write block will execute on the original thread. This means that objects/queries captured before the block can be used inside the block without relying on threadsafe references. 
  * Enumerate all the new API
  * Enumerate the deprecated API
  * Explain the behavior when started on a bg thread (i.e. it will run synchronously)
  * example
```